src/AppMenuButtonGroup.cc

```
@@ -393,8 +398,8 @@ void AppMenuButtonGroup::resetButtons()
 
 void AppMenuButtonGroup::onMenuReadyForSearch()
 {
-    if (!m_lastSearchQuery.isEmpty() && m_searchUiVisible) {
-        filterMenu(m_lastSearchQuery);
+    if (m_search && m_searchUiVisible && m_search->hasValidQuery()) {
```

Clang (14.0.6)

[warning] implicit conversion 'Material::AppMenuSearch *' -> bool

(readability-implicit-bool-conversion)